### PR TITLE
Swift nio 2 dependencies

### PR DESCRIPTION
**Pull request by tachyonics**
*Issue #, if available:* #29 

*Description of changes:* Follow the SwiftNIO migration guide[1]. 

[1] https://github.com/apple/swift-nio/blob/master/docs/migration-guide-NIO1-to-NIO2.md

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -30,9 +30,9 @@ let package = Package(
     ],
     dependencies: [
         .package(url: "https://github.com/IBM-Swift/LoggerAPI.git", .upToNextMajor(from: "1.0.0")),
-        .package(url: "https://github.com/apple/swift-nio.git", from: "1.0.0"),
-        .package(url: "https://github.com/apple/swift-nio-extras.git", from: "0.1.3"),
-        .package(url: "https://github.com/amzn/smoke-http.git", from: "1.0.0"),
```
Comment (asd2734) [inline Package.swift]: What is the reason for moving away from package versions here?

Comment (tachyonics) [reply Package.swift]: SwiftNIO 2 is being actively developed and takes advantage of some optimizations in Swift 5 and later. We will have to move to it sooner or later.

Comment (asd2734) [reply Package.swift]: Oops, sorry. I meant to ask on the transition from 1.0.0 on smoke-http to master branch dependency

Comment (tachyonics) [reply Package.swift]: Oh, I see. Yes, we don't want to do that long term but a SwiftNIO 2 based SmokeHTTP hasn't been tagged yet.

Comment (asd2734) [reply Package.swift]: Gotcha. Thanks for the clarification!

**Comment by wlisac** [comment]
@tachyonics Exciting to see this merged in to master. Any idea when this may be available in a tagged release?

**Pull request merged**